Tests/Fuzzers/FuzzTarget.cpp

```diff
@@ -1,87 +1,76 @@
-#include <iostream>
-
-#include <IPv4Layer.h>
-#include <Packet.h>
 #include <PcapFileDevice.h>
+#include <Packet.h>
+#include <Logger.h>
+#include "DumpToFile.h"
+#include "ReadParsedPacket.h"
 
-#include "Logger.h"
-
-#define TMP_FILEPATH "/tmp/fuzz_sample.pcap"
+static std::string tmpName;
+static std::string tmpFile;
 
-// This function is created as PcapPlusPlus doesn't seem to offer a way of
-// parsing Pcap files directly from memory
-int dumpDataToPcapFile(const uint8_t *data, size_t size)
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
-	FILE *fd;
-	int written = 0;
+	if (tmpName.empty())
+		tmpName = tmpnam (NULL);
 
-	fd = fopen(TMP_FILEPATH, "wb");
-	if (fd == NULL)
-	{
-		std::cerr << "Error opening pcap file for writing\n";
-		return -1;
-	}
+	if (tmpFile.empty())
+		tmpFile = tmpName + FILE_EXT;
 
-	written = fwrite(data, 1, size, fd);
-	if (static_cast<size_t>(written) != size)
+	if (dumpDataToPcapFile(data, size, tmpFile.c_str()) < 0)
 	{
-		std::cerr << "Error writing pcap file\n";
-		fclose(fd);
-		return -1;
+		std::cerr << "Can't Dump buffer to the '" << tmpFile << "' file!!!!\n";
+		return 1;
 	}
 
-	fclose(fd);
-
-	return 0;
-}
+	pcpp::Logger::getInstance().suppressLogs();
 
-extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
-{
-	if (dumpDataToPcapFile(Data, Size) < 0)
+	std::unique_ptr<pcpp::IFileReaderDevice> reader(pcpp::IFileReaderDevice::getReader(tmpFile));
+	if (!reader->open())
 	{
-		std::cerr << "Can't Dump buffer to a PCAP file!!!!\n";
-		return 0;
+		std::cerr << "Error opening the '" << tmpFile << "' file\n";
+		return 1;
 	}
 
-	// Disable logs
-	pcpp::Logger::getInstance().suppressLogs();
+	pcpp::IPcapDevice::PcapStats stats;
+	reader->getStatistics(stats);
+	std::cout << "Read " << stats.packetsRecv << " packets successfully and "
+	                     << stats.packetsDrop << " packets could not be read" << std::endl;
 
-	// open a pcap file for reading
-	pcpp::PcapFileReaderDevice reader(TMP_FILEPATH);
-	if (!reader.open())
+	if (auto ngReader = dynamic_cast<pcpp::PcapNgFileReaderDevice*>(reader.get()))
 	{
-		std::cerr << "Error opening the pcap file\n";
-		return 0;
+		std::cout << "OS is '" << ngReader->getOS() << "'; Hardware is '" << ngReader->getHardware() << "'"
+				  << "'; CaptureApplication is '" << ngReader->getCaptureApplication()
+				  << "'; CaptureFileComment is '" << ngReader->getCaptureFileComment()
+				  << "'" << std::endl;
```

Comment:
As you probably know, pcapng parsing is done using `LightPcapNg` which is a 3rd party library PcapPlusPlus is using. My main concern is that these lines will uncover issues in `LightPcapNg` which we're not familiar with, and not regularly maintain. Please let me know what you think

Reply:
Yes, I looked up and it seems the source code is not there anymore, i.e. PcapPlusPlus has it's own fork of the pcapng parser.
Here is the dilemma: we want to assure our user the library is safe. Users can open pcapng with the library. If there is a bug in the third party library our options are:
a. Fix it in the third party library (not possible in this case?)
b. Fix it in the local fork.
c. Disable pcapng support by default (make a setting that the user has explicitly enable) and bug a big banner that the feature is insecure and experimental.
d. Switch to another library for pcapng handling.

Reply:
I guess we can start with the code as is, if we see too many issues with pcapng we can decide what to do - either options (b) or (c), or just disable opening pcapng files in the fuzzer